crates/router/src/connector/cashtocode.rs

```diff
@@ -0,0 +1,631 @@
+    headers,
+    services::{self, ConnectorIntegration},
+    types::{
+        self,
+        api::{self, ConnectorCommon, ConnectorCommonExt},
+        ErrorResponse, Response,
+    },
+    utils::{self, ByteSliceExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Cashtocode;
+
+impl api::Payment for Cashtocode {}
+impl api::PaymentSession for Cashtocode {}
+impl api::ConnectorAccessToken for Cashtocode {}
+impl api::PreVerify for Cashtocode {}
+impl api::PaymentAuthorize for Cashtocode {}
+impl api::PaymentSync for Cashtocode {}
+impl api::PaymentToken for Cashtocode {}
+impl api::PaymentCapture for Cashtocode {}
+impl api::PaymentVoid for Cashtocode {}
+impl api::Refund for Cashtocode {}
+impl api::RefundExecute for Cashtocode {}
+impl api::RefundSync for Cashtocode {}
+
+fn get_auth_cashtocode(
+    payment_method_data: &api::payments::PaymentMethodData,
+    auth_type: &types::ConnectorAuthType,
+) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+    match payment_method_data {
+        api_models::payments::PaymentMethodData::Reward(reward_data) => {
+            if reward_data.reward_type == "classic" {
+                match auth_type {
+                    types::ConnectorAuthType::BodyKey { api_key, key1: _ } => Ok(vec![(
+                        headers::AUTHORIZATION.to_string(),
+                        format!("Basic {}", api_key.to_owned()),
+                    )]),
+                    _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+                }
+            } else {
+                match auth_type {
+                    types::ConnectorAuthType::BodyKey { api_key: _, key1 } => Ok(vec![(
+                        headers::AUTHORIZATION.to_string(),
+                        format!("Basic {}", key1.to_owned()),
+                    )]),
+                    _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+                }
+            }
+        }
+        _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+    }
+}
+
+impl<Flow, Request, Response> ConnectorCommonExt<Flow, Request, Response> for Cashtocode
+where
+    Self: ConnectorIntegration<Flow, Request, Response>,
+{
+    fn build_headers(
+        &self,
+        req: &types::RouterData<Flow, Request, Response>,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let mut header = vec![(
+            headers::CONTENT_TYPE.to_string(),
+            types::PaymentsAuthorizeType::get_content_type(self).to_string(),
+        )];
+        let default_key = match &req.connector_auth_type {
+            types::ConnectorAuthType::BodyKey { api_key, key1: _ } => api_key.to_owned(),
+            _ => return Err(errors::ConnectorError::FailedToObtainAuthType.into()),
+        };
+        let mut api_key = vec![(headers::AUTHORIZATION.to_string(), default_key)];
+        header.append(&mut api_key);
+        Ok(header)
+    }
+}
```

Comment:
Is this necessary? You are using build_headers in the flows which doesn't have any implementation